src/uint/modular.rs

```diff
@@ -1,7 +1,7 @@
-mod reduction;
-
 /// Implements `Residue`s, supporting modular arithmetic with a constant modulus.
 pub mod constant_mod;
+/// Implements Montgomery reduction
+pub mod reduction;
```

Comment:
Rather than making this module `pub` I think it would make more sense to re-export `montgomery_reduction` from the `modular` module